test/check-subscriptions

```diff
@@ -0,0 +1,232 @@
+#!/usr/bin/python3
+# -*- coding: utf-8 -*-
+
+# This file is part of Cockpit.
+#
+# Copyright (C) 2016 Red Hat, Inc.
+#
+# Cockpit is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# Cockpit is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+
+import os
+import sys
+
+# import Cockpit's machinery for test VMs and its browser test API
+TEST_DIR = os.path.dirname(__file__)
+sys.path.append(os.path.join(TEST_DIR, "common"))
+sys.path.append(os.path.join(os.path.dirname(TEST_DIR), "bots/machine"))
+from testlib import *
+
+# The candlepin image has a lot of demo data preloaded
+# useful info/commands:
+#    Login: doc      password: password
+#    org:   snowwhite
+#
+#    Login: admin    password: admin
+#    org:   admin
+#
+# if you download product files onto the test machine, these will show up as installed products
+# local directory: /etc/pki/product
+# sample products used in the tests (different subscription results for doc and admin):
+# /home/admin/candlepin/generated_certs/6050.pem
+# /home/admin/candlepin/generated_certs/88888.pem
+#
+# to use the candlepin image on a test machine, either add the certificate or
+# allow insecure connections (/etc/rhsm/rhsm.conf -> "insecure = 1")
+#
+# $IP is the ip of the candlepin machine
+#
+# add an activation key to a pool:
+# curl --insecure --request POST --user admin:admin https://$IP:8443/candlepin/activation_keys/ff80808155ca50b10155ca50cd280010/pools/ff80808155ca50b10155ca51f04607e5
+# register with: activation key "awesome_os_pool" and org "admin"
+# or: subscription-manager register --activationkey awesome_os_pool --org admin --serverurl https://$IP:8443/candlepin
+#
+# in order to get the right ids for the activation key and pool, see ACTIVATION_KEY_COMMAND and POOL_COMMAND
+
+WAIT_SCRIPT = """
+set -ex
+for x in $(seq 1 200); do
+    if curl --insecure -s https://%(addr)s:8443/candlepin; then
+        break
+    else
+        sleep 1
+    fi
+done
+"""
```

Comment:
> Calling bash script from python script is unnecessary hack :-)

This script runs in the VM itself, via `self.machine.execute(...)`, so this is not such a big hack as it might seem.

Still, Python is strictly nicer than bash, so I'll rewrite the bash bits.